tox fails with git merge tests

I've got non-passing tox tests for git merge addon.
My `~.gitconfig` contains:
```text
[branch]
	autosetuprebase = always
[pull]
	rebase = true

```
And this causes tests to fail, for example:
```
_________________________________________________________________________________________________ ERROR at setup of test_ignores_binary_files _________________________________________________________________________________________________

tmpdir = local('/tmp/pytest-of-kaszpir/pytest-17/test_ignores_binary_files0')

    @pytest.fixture
    def f1_is_a_conflict_file(tmpdir):
        # Make a merge conflict
        repo1 = tmpdir.join('repo1')
        repo1_f1 = repo1.join('f1')
        repo2 = tmpdir.join('repo2')
        repo2_f1 = repo2.join('f1')
    
        cmd_output('git', 'init', '--', repo1.strpath)
        with repo1.as_cwd():
            repo1_f1.ensure()
            cmd_output('git', 'add', '--', repo1_f1.strpath)
            cmd_output('git', 'commit', '--no-gpg-sign', '-m', 'commit1')
    
        cmd_output('git', 'clone', repo1.strpath, repo2.strpath)
    
        # Commit in master
        with repo1.as_cwd():
            repo1_f1.write('parent\n')
            cmd_output('git', 'commit', '--no-gpg-sign', '-am', 'master commit2')
    
        # Commit in clone and pull
        with repo2.as_cwd():
            repo2_f1.write('child\n')
            cmd_output('git', 'commit', '--no-gpg-sign', '-am', 'clone commit2')
            cmd_output('git', 'pull', retcode=None)
            # We should end up in a merge conflict!
            f1 = repo2_f1.read()
            assert f1.startswith(
                '<<<<<<< HEAD\n'
                'child\n'
                '=======\n'
                'parent\n'
                '>>>>>>>',
            ) or f1.startswith(
                '<<<<<<< HEAD\n'
                'child\n'
                # diff3 conflict style git merges add this line:
                '||||||| merged common ancestors\n'
                '=======\n'
                'parent\n'
                '>>>>>>>',
            ) or f1.startswith(
                # .gitconfig with [pull] rebase = preserve causes a rebase which
                # flips parent / child
                '<<<<<<< HEAD\n'
                'parent\n'
                '=======\n'
                'child\n'
                '>>>>>>>',
            )
>           assert os.path.exists(os.path.join('.git', 'MERGE_MSG'))
E           AssertionError: assert False
E            +  where False = <function exists at 0x7f9d7c31ac80>('.git/MERGE_MSG')
E            +    where <function exists at 0x7f9d7c31ac80> = <module 'posixpath' from '/home/kaszpir/src/pre-commit-hooks/.tox/py36/lib/python3.6/posixpath.py'>.exists
E            +      where <module 'posixpath' from '/home/kaszpir/src/pre-commit-hooks/.tox/py36/lib/python3.6/posixpath.py'> = os.path
E            +    and   '.git/MERGE_MSG' = <function join at 0x7f9d7c2c5598>('.git', 'MERGE_MSG')
E            +      where <function join at 0x7f9d7c2c5598> = <module 'posixpath' from '/home/kaszpir/src/pre-commit-hooks/.tox/py36/lib/python3.6/posixpath.py'>.join
E            +        where <module 'posixpath' from '/home/kaszpir/src/pre-commit-hooks/.tox/py36/lib/python3.6/posixpath.py'> = os.path

tests/check_merge_conflict_test.py:66: AssertionError
=============================================================================================== 321 passed, 2 xfailed, 6 error in 2.55 seconds ================================================================================================
ERROR: InvocationError: '/home/kaszpir/src/pre-commit-hooks/.tox/py36/bin/coverage run -m pytest tests'

```

Quickly checked the directories and seems like there is no .git/MERGE_MSG files.
Status of the tested directories:
```09:11:37 kaszpir@laptop /tmp/pytest-of-kaszpir/pytest-17/test_ignores_binary_files0  $ ll
total 24
drwxrwxr-x   4 kaszpir kaszpir  4096 Mar 26 09:07 ./
drwxrwxr-x 167 kaszpir kaszpir 12288 Mar 26 09:07 ../
drwxrwxr-x   3 kaszpir kaszpir  4096 Mar 26 09:07 repo1/
drwxrwxr-x   3 kaszpir kaszpir  4096 Mar 26 09:07 repo2/
09:11:45 kaszpir@laptop /tmp/pytest-of-kaszpir/pytest-17/test_ignores_binary_files0  $ cd repo1/
09:11:49 kaszpir@laptop /tmp/pytest-of-kaszpir/pytest-17/test_ignores_binary_files0/repo1  (master) $ git log
commit 5313190a8e650e16987321302844fb5a5d353241 (HEAD -> master)
Author: test <test@example.com>
Date:   Mon Mar 26 09:07:32 2018 +0200

    master commit2

commit 9982d02247605a82a4d06dc47df2fb47fcdda642
Author: test <test@example.com>
Date:   Mon Mar 26 09:07:32 2018 +0200

    commit1
09:11:50 kaszpir@laptop /tmp/pytest-of-kaszpir/pytest-17/test_ignores_binary_files0/repo1  (master) $ git status
On branch master
nothing to commit, working tree clean
09:11:53 kaszpir@laptop /tmp/pytest-of-kaszpir/pytest-17/test_ignores_binary_files0/repo1  (master) $ cd ..
09:11:54 kaszpir@laptop /tmp/pytest-of-kaszpir/pytest-17/test_ignores_binary_files0  $ cd repo2
09:12:03 kaszpir@laptop /tmp/pytest-of-kaszpir/pytest-17/test_ignores_binary_files0/repo2  (master|REBASE 1/1) $ git log
commit 5313190a8e650e16987321302844fb5a5d353241 (HEAD, origin/master, origin/HEAD)
Author: test <test@example.com>
Date:   Mon Mar 26 09:07:32 2018 +0200

    master commit2

commit 9982d02247605a82a4d06dc47df2fb47fcdda642
Author: test <test@example.com>
Date:   Mon Mar 26 09:07:32 2018 +0200

    commit1
09:12:06 kaszpir@laptop /tmp/pytest-of-kaszpir/pytest-17/test_ignores_binary_files0/repo2  (master|REBASE 1/1) $ git status
rebase in progress; onto 5313190
You are currently rebasing branch 'master' on '5313190'.
  (fix conflicts and then run "git rebase --continue")
  (use "git rebase --skip" to skip this patch)
  (use "git rebase --abort" to check out the original branch)

Unmerged paths:
  (use "git reset HEAD <file>..." to unstage)
  (use "git add <file>..." to mark resolution)

	both modified:   f1

no changes added to commit (use "git add" and/or "git commit -a")
09:12:08 kaszpir@laptop /tmp/pytest-of-kaszpir/pytest-17/test_ignores_binary_files0/repo2  (master|REBASE 1/1) $ 
```
So we see it was not commited.

--- a/tests/check_merge_conflict_test.py
+++ b/tests/check_merge_conflict_test.py
@@ -37,7 +37,7 @@ def f1_is_a_conflict_file(tmpdir):
     with repo2.as_cwd():
         repo2_f1.write('child\n')
         cmd_output('git', 'commit', '--no-gpg-sign', '-am', 'clone commit2')
-        cmd_output('git', 'pull', retcode=None)
+        cmd_output('git', 'pull', '--no-rebase', retcode=None)
         # We should end up in a merge conflict!
         f1 = repo2_f1.read()
         assert f1.startswith(

